src/main/java/com/atlauncher/mclauncher/MCLauncher.java

```diff
@@ -409,6 +410,63 @@ private static List<String> getArguments(AbstractAccount account, Instance insta
             }
         }
 
+        // Quick Play feature (with backward compatibility for older versions of Minecraft)
+        QuickPlay quickPlay = instance.launcher.quickPlay;
+
+        // Quick Play Multiplayer
+        if (quickPlay.getServerAddress() != null && !quickPlay.getServerAddress().isEmpty()) {
+            String enteredServerAddress = quickPlay.getServerAddress();
+            boolean hasQuickPlayMultiplayer = instance.arguments.game.stream().anyMatch(
+                argumentRule -> argumentRule.value instanceof List &&
+                    ((List<?>) argumentRule.value).contains("--quickPlayMultiplayer")
+            );
+            if (hasQuickPlayMultiplayer) {
+                // Minecraft 23w14a and newer versions
+                arguments.addAll(Arrays.asList("--quickPlayMultiplayer", enteredServerAddress));
+                arguments.add(enteredServerAddress);
+            } else {
+                // Minecraft 23w13a and older versions
+                String[] parts = enteredServerAddress.contains(":") ? enteredServerAddress.split(":")
+                    : new String[]{enteredServerAddress};
+                String address = parts[0];
+                String port = parts.length > 1 ? parts[1] : String.valueOf(Constants.MINECRAFT_DEFAULT_SERVER_PORT);
+                arguments.addAll(Arrays.asList("--server", address));
+                arguments.addAll(Arrays.asList("--port", port));
+            }
+        }
+
+        // Quick Play Single Player
+        if (quickPlay.getWorldName() != null && !quickPlay.getWorldName().isEmpty()) {
+            String selectedWorldSaveName = quickPlay.getWorldName();
+            boolean hasQuickPlaySinglePlayer = instance.arguments.game.stream().anyMatch(
+                argumentRule -> argumentRule.value instanceof List &&
+                    ((List<?>) argumentRule.value).contains("--quickPlaySingleplayer")
+            );
+            if (hasQuickPlaySinglePlayer) {
+                // Minecraft 23w14a and newer versions
+                arguments.addAll(Arrays.asList("--quickPlaySingleplayer", selectedWorldSaveName));
+            } else {
```

Comment:
Now the issue with this and realms is there is no way to do this on older versions, so we can remove the empty else blocks here

It might be good to do these checks in the settings page and disable the singleplayer and realms options in the dropdown if it's not supported so they cannot be selected.

Reply:
> Now the issue with this and realms is there is no way to do this on older versions, so we can remove the empty else blocks here
> 
> It might be good to do these checks in the settings page and disable the singleplayer and realms options in the dropdown if it's not supported so they cannot be selected.

Yes, I thought about that, but then we will end up having only one option for newer versions. Should we do the check using the game args or the game version? We might need a way to compare the versions correctly

Reply:
Game args check like you've done here should be fine since it's only supported on versions that have these params anyway

Reply:
What about the single player, is there a way?

Reply:
Done, doing testing now to make sure it work

Reply:
> It might be good to do these checks in the settings page and disable the singleplayer and realms options in the dropdown if it's not supported so they cannot be selected.

Done, doing testing now to make sure it work

Reply:
Updated in [325d2f42fe6e2ea8bef6979687a50720c01be3f4](https://github.com/ATLauncher/ATLauncher/pull/893/commits/325d2f42fe6e2ea8bef6979687a50720c01be3f4)

While I tested this feature with different cases, we might want to test it more before releasing, I can build the jar for you or you can build it yourself by cloning the fork and the branch for this feature, or you can approve it and then test it before release the new version